packages/SwingSet/src/lib/kdebug.js

```diff
@@ -1,7 +1,27 @@
+import { assert } from '@agoric/assert';
+
 let enableKDebug = false;
+const toggleListeners = new Set();
+
+export function onToggleDebug(callback) {
+  assert.typeof(callback, 'function');
+  toggleListeners.add(callback);
+  callback(enableKDebug);
+}
+
+export function removeToggleDebugListener(callback) {
+  assert.typeof(callback, 'function');
+  return toggleListeners.delete(callback);
+}
 
 export function kdebugEnable(flag) {
-  enableKDebug = !!flag;
+  if (enableKDebug === !!flag) {
+    return;
+  }
+  enableKDebug = !enableKDebug;
+  for (const callback of toggleListeners) {
+    callback(enableKDebug);
+  }
```

Comment:
This whole listeners toggle mechanism is brittle and way too complicated for the job.  It's just a debug flag. so just export a boolean.